### ISSUE TITLE
[iOS]TextField can't delete or insert charactor in the middle of pinyin 

### ISSUE THREAD

**Issue by todoger** [flutter/flutter#86471]
Flutter 2.2.3 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f4abaa0735 (2 weeks ago) • 2021-07-01 12:46:11 -0700
Engine • revision 241c87ad80
Tools • Dart 2.13.4

import 'package:flutter/material.dart';

void main() {
  runApp(FlutterView());
}

class FlutterView extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter View',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  void initState() {
    super.initState();
  }

  final TextEditingController _textEditingController = TextEditingController();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: TextField(
          controller: _textEditingController,
        ),
      ),
    );
  }
}

https://user-images.githubusercontent.com/7564843/125759418-07e38eb6-4ba8-419e-b3b9-dfd3dc0e0847.MP4

**Comment by darshankawar** [comment]
@todoger 
Tried your code sample on latest master using iphone 6s (14.4.1) and used chinese-simplified language which worked properly as below:


https://user-images.githubusercontent.com/67046386/125771284-7c780c1f-bbb4-482f-902e-bf9d7253876a.MP4

<details>
<summary> flutter doctor -v </summary>

```
dhs@Dhss-MacBook-Pro ~ % flutter doctor -v
[✓] Flutter (Channel master, 2.4.0-5.0.pre.55, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.4.0-5.0.pre.55 at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b549e829e8 (11 hours ago), 2021-07-14 15:51:10 -0700
    • Engine revision 093c4ed1a3
    • Dart version 2.14.0 (build 2.14.0-312.0.dev)

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (4 available)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 91.0.4472.114

• No issues found!






```

</details>

**Comment by todoger** [comment]
> @todoger
> Tried your code sample on latest master using iphone 6s (14.4.1) and used chinese-simplified language which worked properly as below:
> 
>  RPReplay-Final1626343756.MP4 
> flutter doctor -v

the problem is that we can't insert charactor in the middle of  the text with underline

https://user-images.githubusercontent.com/7564843/125777237-399594dc-665b-4095-b591-0cc61ba4e4f8.MP4

**Comment by todoger** [comment]
UITextField  in safari
as we expected

https://user-images.githubusercontent.com/7564843/125778259-39513867-5a5c-48f7-abf0-5ba8ecf7d4ae.MP4

**Comment by darshankawar** [comment]
Thanks for the details. I see the same behavior, as below:


https://user-images.githubusercontent.com/67046386/125780887-76f1e6f7-c5a3-482a-a8d9-4fca2f8d1055.MP4

<details>
<summary> flutter doctor -v </summary>

```
[✓] Flutter (Channel master, 2.4.0-5.0.pre.55, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.4.0-5.0.pre.55 at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b549e829e8 (13 hours ago), 2021-07-14 15:51:10 -0700
    • Engine revision 093c4ed1a3
    • Dart version 2.14.0 (build 2.14.0-312.0.dev)

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (4 available)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 91.0.4472.114

• No issues found!






```

</details>

**Comment by huangyumeng123** [comment]
Any update on this problem?

**Comment by justinmc** [comment]
@darshankawar Can you try reproducing this again on the latest master when you have a chance?  This issue looks like it would be a major problem for all iOS Chinese keyboard users, but there doesn't seem to be a lot of traffic here.

**Comment by darshankawar** [comment]
@justinmc Sure.

**Comment by darshankawar** [comment]
Hello @justinmc I verified this issue using latest master on iOS device and observed following:

1. Using Chinese-simplified keyboard, if we input a text with underline (ex: type `chan`, enter space that will convert the word in chinese and type in another text such as `chen`, then move cursor between the entered text and try to type which won't allow to input in the middle but will continue at the end of the entered text as shown below:



https://github.com/flutter/flutter/assets/67046386/fc566c6d-d7a9-47bd-8892-7b968ca026c2

2. In this case, if we let the entered text get converted into Chinese and then try to input in between the text, then it allows the text to be inserted in the middle as shown in same video.

TLDR: as long as the text being input is shown in English, then it doesn't allow any input in the middle. If the text gets converted into Chinese by entering space after the word and then if we try to input in the middle, then it allows to do so.

**Comment by justinmc** [comment]
@darshankawar Thanks for doing that investigation!  Sounds like a very bad experience for users with Chinese keyboards and we should definitely fix this.

**Comment by Renzo-Olivares** [comment]
Closing this issue as a duplicate of https://github.com/flutter/flutter/issues/108016. Please feel free to re-open if this is not the case.
